server/lib/auth.js

```diff
@@ -93,16 +95,53 @@ export function verifyTwoFactorAuthenticatorCode(encryptedTwoFactorAuthToken, tw
   return verified;
 }
 
-export function enforceTwoFactorAuthenticationOnPayouts(req, twoFactorAuthenticatorCode) {
+const getCacheKey = userId => {
+  return `${userId}_2fa_payment_limit`;
+};
+
+export async function rollingPayoutLimitTwoFactorAuthentication(
+  req,
+  twoFactorAuthenticatorCode,
+  hostRollingLimit,
+  expenseAmount,
+) {
   if (req.remoteUser.twoFactorAuthToken !== null) {
+    // 1. we check the 'cache' if the key exists: cacheKey=${user.id}_2fa_payment_limit
+    const userId = req.remoteUser.id;
+    const cacheKey = getCacheKey(userId);
+    const keyInCache = await cache.get(cacheKey);
```

Comment:
Use a more describing variable name (ideally without cache), as it's used later in `const runningTotal = keyInCache + expenseAmount;`